src/hotspot/os/windows/os_windows.cpp

```diff
@@ -558,7 +567,8 @@ unsigned __stdcall os::win32::thread_native_entry(void* t) {
 
   // Thread must not return from exit_process_or_thread(), but if it does,
   // let it proceed to exit normally
-  return (unsigned)os::win32::exit_process_or_thread(os::win32::EPT_THREAD, res);
+  exit_process_or_thread(EPT_THREAD, res);
+  return res;
```

Comment:
It has to call `exit_process_or_thread` because of the exit bug. That means we will directly call `_endthreadex` instead of implicitly doing that by having the entry function return. The `return res` is redundant but keeps the compiler happy given the required declaration of the entry method.

Reply:
I'm not too sure what to make of this, since I don't know what the exit bug is about (Also, the return res doesn't cause an issue on MSVC under any circumstance, and would only do so on gcc if thread_native_entry was marked noreturn, which it isn't). I simply changed the return value to keep the original semantics of the code unchanged, I guess I should take this to mean I should keep my current changes as is?

Reply:
Yes, keep as is in this area.